buildSrc/src/main/java/org/elasticsearch/gradle/JdkDownloadPlugin.java

```diff
@@ -0,0 +1,187 @@
+                    final ConfigurationContainer configurations = project.getConfigurations();
+                    String configurationName = configName("localjdk", version, platform);
+                    Configuration jdkConfig = configurations.findByName(configurationName);
+                    if (jdkConfig == null) {
+                        jdkConfig = configurations.create(configurationName);
+                    }
+                    task.dependsOn(jdkConfig);
+
+                    // depend on the jdk directory "artifact" from the root project
+                    DependencyHandler dependencies = project.getDependencies();
+                    Map<String, Object> depConfig = new HashMap<>();
+                    depConfig.put("path", ":"); // root project
+                    depConfig.put("configuration", configName("localjdk", version, platform));
+                    dependencies.add(configurationName, dependencies.project(depConfig));
+
+                    // add jdkHome runtime property to the task
+                    Object jdkHomeGetter = new Object() {
+                        @Override
+                        public String toString() {
+                            return configurations.getByName(configurationName).getSingleFile().toString();
+                        }
+                    };
+                    extraProperties.set("jdkHome", jdkHomeGetter);
+                }
+            })
+        );
+    }
+
+    private static void setupRootJdkDownload(Project rootProject, String platform, String version) {
+        String extractTaskName = "extract" + capitalize(platform) + "Jdk" + version;
+        if (rootProject.getTasks().findByName(extractTaskName) != null) {
+            // already setup this version
+            return;
+        }
+
+        // decompose the bundled jdk version, broken into elements as: [feature, interim, update, build]
+        // Note the "patch" version is not yet handled here, as it has not yet been used by java.
+        Matcher jdkVersionMatcher = JDK_VERSION.matcher(version);
+        if (jdkVersionMatcher.matches() == false) {
+            throw new IllegalArgumentException("Malformed jdk version [" + version + "]");
+        }
+        String jdkVersion = jdkVersionMatcher.group(1) + (jdkVersionMatcher.group(2) != null ? (jdkVersionMatcher.group(2)) : "");
+        String jdkMajor = jdkVersionMatcher.group(1);
+        String jdkBuild = jdkVersionMatcher.group(3);
+
+        // add fake ivy repo for jdk url
+        String repoName = "jdk_repo_" + version;
+        if (rootProject.getRepositories().findByName(repoName) == null) {
+            rootProject.getRepositories().ivy(ivyRepo -> {
+                ivyRepo.setName(repoName);
+                ivyRepo.setUrl("https://download.java.net");
+                ivyRepo.patternLayout(layout -> {
+                    layout.artifact("java/GA/jdk" + jdkMajor + "/" + jdkBuild + "/GPL/openjdk-[revision]_[module]-x64_bin.[ext]");
```

Comment:
we could add something like: 
```
spec.content(c-> c.includeGroup("org.elasticsearch.jdk-downloader."));
```
And use "org.elasticsearch.jdk-downloader" as a group when downloading the dependency. 
It's a good practice for repositories confirmed by plugins.

Reply:
Sorry I don't follow. The group is completely arbitrary and internal since this is a fake ivy repo.

Reply:
But the repository is not, so all dependencies could potentially be checked against it. 
I just suggested `org.elasticsearch.jdk-downloader` to make it more specific, or make it "ours" since it's not really used in the pattern layout. It would work the same way with "jdk".